### PR TITLE
Implement Town Screen Hotkeys

### LINKED ISSUE #3640: Refactor and Optimize processing for Castle Dialog
In `castle_dialog.cpp` I found there are two for loops using `cacheBuildings` in which both of the loops seem to loop through the same things.

`msg_status = Monster( race, ( *it ).id ).GetName();`
Also, this code does nothing since `msg_status` will be replaced with the next loop. The result is also not correct in the first place. The message should be `Recruit Peasants` which is correctly returned by the other loop and not just `Peasant`.

I am making this issue since I will be making changes to these logics for town and build screen hotkeys so please assign this issue to me. Also, I want to check if there anyone else currently editing this file at the moment to prevent us both making the same changes to the file :)

### PR DESCRIPTION
**Pull request by eos428**
Also adds hotkeys as discussed in #3438. Town screen only at the moment.
I'd like to apply the hotkeys in build screen too but probably it's better to do it in a different PR.

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @eos428 , could you please fix compilation on Windows (AppVeyor)?

**Comment by ihhub** [comment]
Hi @Branikolog , could you please test these changes?

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @eos428 , I left just one comment for now. Could you please check it?

File: src/fheroes2/castle/castle_dialog.cpp [hunk 1]
```diff
@@ -47,6 +47,54 @@
 
 void CastleRedrawTownName( const Castle & castle, const fheroes2::Point & dst );
 
+namespace
+{
+    auto getPressedBuildingHotkey() -> building_t
```
Comment (ihhub) [inline src/fheroes2/castle/castle_dialog.cpp:52]: Why can't we just do `building_t getPressedBuildingHotkey()` ?

File: src/fheroes2/castle/castle_dialog.cpp [hunk 2]
```diff
@@ -375,195 +423,201 @@ int Castle::OpenDialog( bool readonly )
...
-                        fheroes2::ButtonRestorer exitRestorer( buttonExit );
-                        if ( Castle::RecruitMonster(
-                                 Dialog::RecruitMonster( Monster( race, GetActualDwelling( ( *it ).id ) ), getMonstersInDwelling( ( *it ).id ), true ) ) ) {
-                            need_redraw = true;
-                        }
-                    }
-                    else if ( le.MousePressRight( ( *it ).coord ) ) {
-                        Dialog::DwellingInfo( Monster( race, GetActualDwelling( ( *it ).id ) ), getMonstersInDwelling( ( *it ).id ) );
-                    }
+            for ( auto it = cacheBuildings.cbegin(); it != cacheBuildings.cend(); ++it ) {
```
Comment (ihhub) [inline src/fheroes2/castle/castle_dialog.cpp:426]: Hi @eos428 , my apologies for a very late request but could we please separate two changes into 2 pull requests? I feel it'd be easier to review with much smaller changes.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @eos428 , I left 2 comments. Would you mind to take a look?

File: src/fheroes2/castle/castle_dialog.cpp (hunk 1 above)
Comment (eos428) [reply src/fheroes2/castle/castle_dialog.cpp:52]: Back then there was a nagging from clang_tidy. I'll revert this change soon

File: src/fheroes2/castle/castle_dialog.cpp (hunk 2 above)
Comment (eos428) [reply src/fheroes2/castle/castle_dialog.cpp]: Hi @ihhub, it's alright. Going forward with this PR I'll remove the refactoring for now. It'll be up again in the PR after this :)

**Comment by ihhub** [comment]
@eos428 , thank you very much for this improvement!

**Pull request merged**